lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -502,6 +502,27 @@ void PrimIfOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
   });
 }
 
+//===----------------------------------------------------------------------===//
+// RuntimeAssertOp
+//===----------------------------------------------------------------------===//
+
+void RuntimeAssertOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
+                                                  MLIRContext *context) {
+  patterns.add(+[](RuntimeAssertOp op, PatternRewriter &rewriter) {
+    bool value;
+    if (!matchPattern(op.getCondition(), m_TorchConstantBool(&value)))
+      return failure();
+
+    if (value) {
+        rewriter.eraseOp(op);
+        return success();
+    }
+    // TODO: If we statically know that the condition is false, should we
```

Comment:
I also wonder if we shouldn't just `op->emitError()` when this happens, so we give a clearer message of why we are failing here instead of later.

Reply:
I think the problem is that op->emitError() doesn't stop the compiler, so we would continue to spam the console with error messages every time that this fold is tried.